src/screens/DiscoveryScreenV2/fragment/UserFragmentCommunity.js

```diff
@@ -0,0 +1,302 @@
+      index
+    });
+  };
+
+  const renderUsersItem = () => {
+    const initFollowingUsers = [];
+    const initUnfollowingUsers = [];
+
+    users.forEach((item) => {
+      if (item.user?.user_id_follower || item.user_id_follower) {
+        initFollowingUsers.push(item);
+      } else {
+        initUnfollowingUsers.push(item);
+      }
+    });
+
+    const data = isFirstTimeOpen
+      ? withoutRecent
+        ? initialUsers.length !== 0
+          ? initialUsers
+          : [...initFollowingUsers, {separator: true}, ...initUnfollowingUsers]
+        : [...initFollowingUsers, {separator: true}, ...initUnfollowingUsers]
+      : unfollowedUsers.length !== 0
+      ? [...followedUsers, {separator: true}, ...unfollowedUsers]
+      : followedUsers;
+
+    return (
+      <FlatList
+        onMomentumScrollBegin={handleScroll}
+        contentContainerStyle={{paddingBottom: 100}}
+        data={data || []}
+        renderItem={renderItem}
+        keyExtractor={(item, index) => index.toString()}
+        onEndReached={() => fetchData()}
+        onEndReachedThreshold={0.6}
+        keyboardDismissMode="on-drag"
+        keyboardShouldPersistTaps="handled"
+        style={{backgroundColor: COLORS.almostBlack}}
+      />
+    );
+  };
+
+  if (isLoadingDiscoveryUser)
+    return (
+      <View style={styles.fragmentContainer}>
+        <LoadingWithoutModal />
+      </View>
+    );
+  if (followedUsers.length === 0 && unfollowedUsers.length === 0 && !isFirstTimeOpen)
+    return (
+      <View style={styles.noDataFoundContainer}>
+        <Text style={styles.noDataFoundText}>No users found</Text>
+      </View>
+    );
+
+  return <View>{renderUsersItem()}</View>;
+};
+
+const styles = StyleSheet.create({
+  fragmentContainer: {
+    flex: 1,
+    backgroundColor: COLORS.almostBlack
+  },
+  noDataFoundContainer: {
+    flex: 1,
+    backgroundColor: COLORS.almostBlack,
+    justifyContent: 'center'
+  },
+  noDataFoundText: {
+    alignSelf: 'center',
+    justifyContent: 'center',
+    fontFamily: fonts.inter[600],
+    color: COLORS.white
+  },
+  unfollowedHeaderContainer: {
+    backgroundColor: COLORS.gray110,
+    height: 40,
+    display: 'flex',
+    flexDirection: 'column',
+    justifyContent: 'center'
+  },
+  unfollowedHeaders: {
+    fontFamily: fonts.inter[600],
+    marginLeft: 20
+  },
+  containerHidden: {
+    display: 'none'
+  }
+});
+
+UsersFragmentCommunity.propTypes = {
+  isLoadingDiscoveryUser: PropTypes.bool,
+  isFirstTimeOpen: PropTypes.bool,
+  initialUsers: PropTypes.array,
+  setInitialUsers: PropTypes.func,
+  followedUsers: PropTypes.array,
+  setFollowedUsers: PropTypes.func,
+  unfollowedUsers: PropTypes.array,
+  setUnfollowedUsers: PropTypes.func,
+  setSearchText: PropTypes.func,
+  setIsFirstTimeOpen: PropTypes.func,
+  withoutRecent: PropTypes.bool,
+  showRecentSearch: PropTypes.bool,
+  fetchData: PropTypes.func,
+  searchText: PropTypes.string
+};
+
+export default UsersFragmentCommunity;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The code looks good overall, but there are a few areas that could be improved for better readability and maintainability.

Firstly, the `handleFollow` function (lines 91-115) is doing too much. It's handling user following/unfollowing, making API calls, and also managing error handling. This function could be broken down into smaller functions each with a single responsibility. 

Secondly, the `renderItem` function (lines 167-197) has complex conditional logic to determine the `from` value. This could be simplified or extracted into a separate function for clarity.

Lastly, the `renderUsersItem` function (lines 199-234) is preparing data for rendering and also returning the JSX. These responsibilities could be separated for better readability.

Here are the suggested changes:

```diff
+ const handleUserFollowing = async (from, willFollow, item) => {
+   // ...existing code...
+ };
+ 
+ const makeFollowApiCall = async (willFollow, data) => {
+   if (willFollow) {
+     await setFollow(data, client);
+   } else {
+     await setUnFollow(data, client);
+   }
+ };
+ 
+ const handleFollow = async (from, willFollow, item) => {
+   // ...existing code...
+   try {
+     await makeFollowApiCall(willFollow, data);
+   } catch (error) {
+     handleUserFollowing(from, !willFollow, item);
+   }
+   // ...existing code...
+ };
  
+ const getFromValue = (index) => {
+   let result;
+   // ...existing code...
+   return result;
+ };
+ 
+ const renderItem = ({index, item}) => {
+   const from = getFromValue(index);
+   return renderDiscoveryItem({from, item, index});
+ };
  
+ const prepareUsersData = () => {
+   // ...existing code...
+   return data;
+ };
+ 
+ const renderUsersItem = () => {
+   const data = prepareUsersData();
+   // ...existing code...
+ };
```

These changes should make the code easier to read and maintain. Each function now has a single responsibility which follows the Single Responsibility Principle (SRP) of SOLID principles.


<!-- This is an auto-generated comment by OSS CodeRabbit -->